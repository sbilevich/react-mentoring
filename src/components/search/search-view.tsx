import { AppButton } from 'components/app-button/app-button';
import { FC, useEffect, useState } from 'react';

import styles from './search.module.scss';

export interface SearchViewProps {
  onSearch: (searchText: string) => void;
  searchQuery?: string;
}

const placeholder = 'What do you want to watch?';

export const SearchView: FC<SearchViewProps> = ({ onSearch, searchQuery }) => {
  const [searchText, setSearchText] = useState<string>('');

  useEffect(() => {
    if (searchQuery) {
      setSearchText(searchQuery);
    }
  }, [searchQuery]);

  return (
    <div className={styles.searchWrapper}>
      <input
        type="text"
        className={styles.input}
        placeholder={placeholder}
        value={searchText}
        onChange={(e) => setSearchText(e.target.value)}
      />
      <AppButton text="Search" onButtonClick={() => onSearch(searchText)} />
    </div>
  );
};
