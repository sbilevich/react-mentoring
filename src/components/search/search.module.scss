@import "../../palette.scss";

.searchWrapper {
  display: flex;
}

.input {
  border: none;
  padding: 0 0 0 19px;
  margin-right: 14px;
  width: 715px;
  height: 57px;
  background: rgba(50, 50, 50, 0.8);
  border-radius: 4px;
  font-weight: 400;
  font-size: 20px;
  line-height: 24px;
  color: $white;
}
