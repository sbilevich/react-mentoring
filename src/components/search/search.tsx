import { AppButton } from 'components/app-button/app-button';
import { FC, useEffect, useState } from 'react';
import { useNavigate, useParams } from 'react-router-dom';
import { useAppDispatch } from 'redux/hooks';
import { fetchMoviesAction } from 'redux/movies';

import styles from './search.module.scss';

const placeholder = 'What do you want to watch?';

export const Search = () => {
  const { searchQuery } = useParams();

  const navigate = useNavigate();

  useEffect(() => {
    if (searchQuery) {
      dispatch(
        fetchMoviesAction({ searchText: searchQuery, searchBy: 'title' }),
      );
    }
  }, [searchQuery]);

  const dispatch = useAppDispatch();

  const handleSearch = (searchText: string) => {
    navigate(`/search/${searchText}`);
    dispatch(fetchMoviesAction({ searchText, searchBy: 'title' }));
  };

  return <SearchView onSearch={handleSearch} searchQuery={searchQuery} />;
};

export interface SearchViewProps {
  onSearch: (searchText: string) => void;
  searchQuery?: string;
}
export const SearchView: FC<SearchViewProps> = ({ onSearch, searchQuery }) => {
  const [searchText, setSearchText] = useState<string>('');

  useEffect(() => {
    if (searchQuery) {
      setSearchText(searchQuery);
    }
  }, [searchQuery]);

  return (
    <div className={styles.searchWrapper}>
      <input
        type="text"
        className={styles.input}
        placeholder={placeholder}
        value={searchText}
        onChange={(e) => setSearchText(e.target.value)}
      />
      <AppButton text="Search" onButtonClick={() => onSearch(searchText)} />
    </div>
  );
};
