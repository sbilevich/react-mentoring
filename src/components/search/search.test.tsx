import { screen, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { SearchView } from './search';

const noop = jest.fn();

describe('Search', () => {
  test('renders SearchSnapshot', () => {
    const { asFragment } = render(<SearchView onSearch={noop} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should call on search with userText', () => {
    const onSearch = jest.fn();
    render(<SearchView onSearch={onSearch} />);
    const input = screen.getByRole('textbox');
    userEvent.type(input, 'test');
    const button = screen.getByRole('button');
    userEvent.click(button);

    expect(onSearch).toHaveBeenCalledWith('test');
  });

  it('should set value to input if searchQuery', async () => {
    render(<SearchView onSearch={noop} searchQuery="testQuery" />);
    const inputText = screen.queryByDisplayValue('testQuery');

    expect(inputText).toBeInTheDocument();
  });
});
