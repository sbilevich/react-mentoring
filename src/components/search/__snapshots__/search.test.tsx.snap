// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders SearchSnapshot 1`] = `
<DocumentFragment>
  <div
    class="searchWrapper"
  >
    <input
      class="input"
      placeholder="What do you want to watch?"
      type="text"
      value=""
    />
    <button
      class="button"
      type="button"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
